### PR TITLE
refactor(iot-service): Model Id changes for Twin

### PR DESCRIPTION
**Pull request by bikamani**
Property name is now changed to "modelId"
Tested and verified with Sample.

### CODE REVIEW

File: shared/src/TwinJsonConverter.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public sealed class TwinJsonConverter : JsonConverter
         private const string CloudToDeviceMessageCountTag = "cloudToDeviceMessageCount";
         private const string AuthenticationTypeTag = "authenticationType";
         private const string X509ThumbprintTag = "x509Thumbprint";
-        private const string ModelId = "digital-twin-model-id";
+        private const string ModelId = "modelId";
```
Comment (abhipsaMisra) [inline shared/src/TwinJsonConverter.cs:42]: so this is `modelId` and not `model-id`?

Comment (bikamani) [reply shared/src/TwinJsonConverter.cs:42]: yes

File: shared/src/TwinJsonConverter.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public sealed class TwinJsonConverter : JsonConverter
...
 
         /// <summary>
```
Comment (prmathur-microsoft) [inline shared/src/TwinJsonConverter.cs:44]: Will there be separate PR for Device Client?

Comment (bikamani) [reply shared/src/TwinJsonConverter.cs:44]: Separate [PR ](https://github.com/Azure/azure-iot-sdk-csharp/pull/1431)because it is for master branch, merged just a minute ago.

File: shared/src/TwinJsonConverter.cs (hunk 1 above)
Comment (prmathur-microsoft) [reply shared/src/TwinJsonConverter.cs:42]: https://github.com/Azure/azure-iot-sdk-java/pull/835/files as per this it model-id. So is it different between device client and service client?

Comment (prmathur-microsoft) [reply shared/src/TwinJsonConverter.cs:42]: Okay this resolves it. https://github.com/Azure/azure-iot-sdk-java/pull/834/files

File: shared/src/TwinJsonConverter.cs (hunk 2 above)
Comment (prmathur-microsoft) [reply shared/src/TwinJsonConverter.cs:44]: 👍
